tests/robustness/engine/engine.go

```diff
@@ -264,3 +275,59 @@ func (e *Engine) init(ctx context.Context) error {
 
 	return e.Checker.VerifySnapshotMetadata()
 }
+
+// InitS3WithServer intializes the Engine with InitS3 and with addition to that it uses server/client model
+func (e *Engine) InitS3WithServer(ctx context.Context, bucketName, testRepoPath, metaRepoPath, addr string) error {
+
+	err := e.MetaStore.ConnectOrCreateS3WithServer(addr, bucketName, metaRepoPath)
```

Comment:
Any reason the metadata repo connection needs to be via client/server as well? Would it be simpler to just connect directly?